### PR TITLE
Jupyter Kernel

### PR DESCRIPTION
**Pull request by Vipul-Cariappa**
#2756

### CODE REVIEW

**Comment by Vipul-Cariappa** [comment]
@certik, I have squash merged the changes make in #2762 and `cherry-pick`ed the necessery changes from #2756.

File: src/lpython/python_evaluator.cpp [hunk 1]
```diff
@@ -156,11 +157,11 @@ Result<PythonCompiler::EvalResult> PythonCompiler::evaluate(
                                     ->m_symtab->get_symbol(run_fn);
             LCOMPILERS_ASSERT(fn)
             if (ASRUtils::get_FunctionType(fn)->m_return_var_type->type == ASR::ttypeType::UnsignedInteger) {
-                uint16_t r = e->execfn<uint16_t>(run_fn);
+                uint16_t r = e->execfn<int>(run_fn);
                 result.type = EvalResult::unsignedInteger2;
                 result.u32 = r;
             } else {
-                int16_t r = e->execfn<int16_t>(run_fn);
+                int16_t r = e->execfn<int>(run_fn);
```
Comment (certik) [inline src/lpython/python_evaluator.cpp:164]: Do you know why the previous code was incorrect?

**Comment by certik** [comment]
Looks like there is some issue with the bit_length implementation (https://github.com/lcompilers/lpython/issues/1836), so it's fine to disable the test. The other macOS interactive issue was fixed. I don't understand exactly the problem, but it looks like something with signed/unsigned. The change in this PR works on all platforms, so it gets us unblocked.

Let's merge it, and we can fix the rest of the issues later.

File: ci/environment.yml [hunk 2]
```diff
@@ -4,20 +4,19 @@ channels:
...
+  - pytest
+  - jupyter
+  - jupyter_kernel_test
+  - re2c
+  - numpy
   - zlib
   - zstd
-  - ninja=1.11.0
-  - rapidjson=1.1.0
+  - ninja
```
Comment (certik) [inline ci/environment.yml:20]: We have to put exact version numbers here.

**Comment by certik** [comment]
I am merged this as is, since all tests pass. Now I'll submit a PR to fix the package versions.

File: ci/environment.yml (hunk 2 above)
Comment (certik) [reply ci/environment.yml:20]: Fixed in https://github.com/lcompilers/lpython/pull/2765.

**Pull request merged**